### ISSUE TITLE
MedicationRequest: Is 'intent' missing from our L3 code?

### ISSUE THREAD

**Issue by ssebast2** [ccsm-cds-tools/ccsm-cds-with-tests#14]
The Intent data element seems to have a 1...1 cardinality, which means it will almost always (or even always) be required/included. 

The value = 'Order' seems most appropriate. 

There seems to be a number of sub-categories of 'Order'. Please let us know if you feel that sub-categories should be included or if you disagree with 1) including Intent, and 2) constraining the value to 'Order'.

**Comment by mickohanlon23** [comment]
There is currently no query of a `MedicationRequest`'s`intent` element in our code. However, we do put a constraint on the `status` of a medicationRequest, limiting it to `completed`.

I don't have much experience working with MedicationRequests, but U.S Core provides helpful documentation here: http://build.fhir.org/ig/HL7/US-Core/medication-list.html#get-all-medications.

I believe we would want to be following the "Get All Medications" instructions, which make it seem like limiting `intent` to `order` and `plan` would be appropriate. Looking at the sub-category explanations [here](http://hl7.org/fhir/R4/valueset-medicationrequest-intent.html#expansion), it seems like the four sub-categories of order would also be appropriate to include.

**Comment by mickohanlon23** [comment]
@ssebast2 - Once you discuss with Rose, could you let me know:
1. Whether I should add a constraint to MedicationRequest.intent
2. If so, [what elements](http://hl7.org/fhir/R4/codesystem-medicationrequest-intent.html) you would like to constrain the data element to.

**Comment by mickohanlon23** [comment]
Thanks for the explanation. Restricting `intent` and leaving out the `plan` code makes sense. It seems like the sub-categories of `order` should also be included, since they just specify the details of the order. Are there any reasons to exclude a MedicationRequest if they have one of the sub-category values in the `intent` field? Do you agree that we should restrict MedicationRequest to:
- order
- original-order
- reflex-order
- filler-order
- instance-order

**Comment by ssebast2** [comment]
@mickohanlon23  - yes, it is fin to include the child attributes under 'order'. Thanks!

**Comment by mickohanlon23** [comment]
I added the `intent` restriction in commit: https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/commit/c4b3af3b11477ba25aef14f031e1f069e68b50ec. I also added testing to make sure that a MedicationRequest with an intent of order and with an intent of one of the order sub-categories are both included, as well as a test to make sure that a MedicationRequest with an intent of proposal is not included.

I'll go ahead and resolve this issue.
